### PR TITLE
Integrated Imgur with authentication

### LINKED ISSUE #776: Adding Imgur login functionality with user authentication
**Actual Behaviour**
Currently, you can upload an image on Imgur anonymously.

**Expected Behaviour**
Will implement user authentication and upload image on Imgur from an account.

**Steps to reproduce it**
Share -> imgur -> add account -> upload from account

**Would you like to work on the issue?**

Yes.

### PR DESCRIPTION
**Pull request by mohitmanuja**
Fix #776 

Changes:

Added Imgur authentication. 
To check -> Accounts imgur -> login with your crendentials.

There will be a Toast message that Imgur account is successfully signed in.

### CODE REVIEW

File: app/src/main/java/org/fossasia/phimpme/sharedimgur/ImgurOuthActivity.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import butterknife.BindView;
+import butterknife.ButterKnife;
+
+import static org.fossasia.phimpme.utilities.Constants.IMGUR_LOGIN_URL;
+
+/**
+ * Created by manuja on 16/7/17.
+ */
+
+public class ImgurOuthActivity extends Activity {
```
Comment (pa1pal) [inline app/src/main/java/org/fossasia/phimpme/sharedimgur/ImgurOuthActivity.java]: @mohitmanuja rename the class name either to `ImgurOAuthActivity` or `ImgurAuthActivity`

File: app/src/main/java/org/fossasia/phimpme/sharedimgur/ImgurOuthActivity.java [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            public void onPageStarted(WebView view, String url, Bitmap favicon) {
+                super.onPageStarted(view, url, favicon);
+
+                if (mAlertBuilder != null) {
+                    mAlertBuilder.show();
+                }
+            }
+        });
+
+
```
Comment (pa1pal) [inline app/src/main/java/org/fossasia/phimpme/sharedimgur/ImgurOuthActivity.java]: remove unnecessary new lines.

**Comment by mohitmanuja** [comment]
@pa1pal I have made the changes please review it again. Thanks !

**Comment by vinaysajjanapu** [comment]
@mohitmanuja resolve conflicts

**Comment by mohitmanuja** [comment]
@vinaysajjanapu I resolved the conflicts please review it.

**Pull request merged**